### PR TITLE
Add registration server

### PR DESCRIPTION
**Pull request by imobachgs**
Add a fake registration server to check:

* Certificates handling
* Self-update SCC/SMT integration

It can be extended to support registration.

### CODE REVIEW

File: lib/aytests/certs_factory.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    SERIAL_LIMIT = 9999
+    # Certificates expiration time (in seconds)
+    EXPIRATION   = 3600
+
+    # Constructor
+    #
+    # @param ca_cert_ path [Pathname] Path to CA certificate
+    # @param ca_key_path   [Pathname] Path to CA certificate
+    def initialize(ca_crt_path:, ca_key_path:)
+      @ca_crt = OpenSSL::X509::Certificate.new(ca_crt_path.read)
```
Comment (schubi2) [inline lib/aytests/certs_factory.rb]: Do we have to add any additional requirements (gem installation while setup the test environment, documentation) here ?

**Comment by schubi2** [comment]
Great job! I am not sure if we have had to use the real SCC server for testing instead in order to find interface errors with SCC at once. But that would not fulfill the requirements for SMT (I assume). Anyway this way does not exclude the other one :-). So really nice ! Lets try it. Please have a look to jenkins if all required gems are installed. ----> LGMT

File: lib/aytests/certs_factory.rb (hunk 1 above)
Comment (imobachgs) [inline lib/aytests/certs_factory.rb]: No, fortunately it's included in Ruby's stdlib

**Comment by imobachgs** [comment]
@schubi2 Thanks :)

You're right that we would need to use SCC at some point to try the "real" stuff. This PR will help to cover another scenario: when you have your own registration server (say SMT) with your own certificates.

Let's give it a try! :)

**Pull request merged**